File: ChibuzorTasks/bankapp/src/bankapp/Account.java
package bankapp.src.bankapp;

public class Account {
    private double balance;
    private String pin;
    private String accountNumber;
    private String firstName;
    private String lastName;

    public Account(String firstName, String lastName, String accountNumber, String pin) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.accountNumber = accountNumber;
        this.pin = pin;

    }

    public String getAccountNumber() {
        return accountNumber;
    }

    public void deposit(double amount) {
        if (amount <= 0) throw new IllegalArgumentException("Amount must be positive");
        this.balance += amount;
    }

    public double getBalance() {
        return balance;
    }

    public void withdraw(double amount, String pin) {
        if (amount <= 0) throw new IllegalArgumentException("Amount must be positive");
        if (balance < amount) throw new IllegalArgumentException("Insufficient balance");
        if(!pin.equals(this.pin)) throw new IllegalArgumentException("Invalid pin");
        this.balance -= amount;

    }

    public void updatePin(String oldPin, String newPin) {
        if(!oldPin.equals(this.pin)) throw new IllegalArgumentException("Invalid pin");
        this.pin = newPin;
    }

    @Override
    public String toString() {
        return "FullName: " +firstName+" "+lastName+ ", accountNumber=" + accountNumber;
    }

}

File: PythonClass/classWorks/Feb-2025/mybank/account.py
class Account:
    def __init__(self, first_name, last_name, account_number, pin):
        self.first_name = first_name
        self.last_name = last_name
        self.account_number = account_number
        self.pin = pin
        self.balance = 0.0

    def get_account_number(self):
        return self.account_number

    def get_pin(self):
        return self.pin

    def deposit(self, amount):
        if amount <= 0:
            raise ValueError("Amount must be positive")
        self.balance += amount

    def get_balance(self):
        return self.balance

    def withdraw(self, amount, pin):
        if amount <= 0:
            raise ValueError("Amount must be positive")
        if self.balance < amount:
            raise ValueError("Insufficient balance")
        if pin != self.pin:
            raise ValueError("Invalid pin")
        self.balance -= amount

    def update_pin(self, old_pin, new_pin):
        if old_pin != self.pin:
            raise ValueError("Invalid pin")
        self.pin = new_pin

    def __str__(self):
        return f"FullName: {self.first_name} {self.last_name}, accountNumber={self.account_number}"
